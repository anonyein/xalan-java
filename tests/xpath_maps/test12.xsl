<?xml version="1.0" encoding="UTF-8"?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
                xmlns:xs="http://www.w3.org/2001/XMLSchema"
                xmlns:map="http://www.w3.org/2005/xpath-functions/map"
				exclude-result-prefixes="xs map"
                version="3.0">				

   <!-- Author: mukulg@apache.org -->                 				

   <xsl:output method="xml" indent="yes"/>
   
   <!-- An XPath 3.1 test case, to test SequenceType MapTest specified as 
        value of xsl:variable's "as" attribute. -->

   <xsl:template match="/">	  
      <result>
	     <xsl:variable name="map1" select="map {'zero' : xs:boolean(0), 'one' : xs:boolean(1)}" as="map(xs:string, xs:boolean)"/>		 
	     <map noOfEntries="{count(map:keys($map1))}">
            <xsl:for-each select="map:keys($map1)">
			   <xsl:variable name="key" select="."/>
			   <entry key="{$key}" value="{map:get($map1, $key)}"/>
			</xsl:for-each>
		 </map>		 
		 <xsl:variable name="map2" select="map {1 : 'Mukul', 2 : 'Joseph', 3 : 'Gary', 4 : 'John', 5 : 'Henry'}" as="map(xs:integer, xs:string)"/>
		 <map noOfEntries="{count(map:keys($map2))}">
            <xsl:for-each select="map:keys($map2)">
			   <xsl:variable name="key" select="."/>
			   <entry key="{$key}" value="{map:get($map2, $key)}"/>
			</xsl:for-each>
		 </map>
	  </result>
   </xsl:template>

</xsl:stylesheet>
