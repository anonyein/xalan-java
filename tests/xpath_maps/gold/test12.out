<?xml version="1.0" encoding="UTF-8"?><result>
  <map noOfEntries="2">
    <entry value="false" key="zero"/>
    <entry value="true" key="one"/>
  </map>
  <map noOfEntries="5">
    <entry value="Mukul" key="1"/>
    <entry value="Joseph" key="2"/>
    <entry value="Gary" key="3"/>
    <entry value="John" key="4"/>
    <entry value="Henry" key="5"/>
  </map>
</result>
